src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -262,8 +293,10 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
                      GradType& gradient,
                      const size_t /* batchSize */)
```

Comment:
(This comment might be out of scope for this PR---I'm not so familiar with the GAN code, so please excuse the questions...)  Why are we ignoring `batchSize` here?  Shouldn't we use the batch size requested by the optimizer?

Reply:
Okay, I was also quite unsure of this.
Just some important points. 
1. The batch size returned by optimizer is the `batchsize` of real data. We take that amount of real data while training in next step. Commenting this means that  the size of input `real data (traindata or predictors)` is always a multiple of `batch size`. I thought that it is always assumed. So I didn't changed it.
Let me know if I should change it.

Reply:
Ok, I think I understand.  In any case, so long as we are respecting the `batchSize` requested by the user when they call `Train(OptimizerType&)`, it sounds good to me.